mule-extensions-archetype-it/src/test/java/org/mule/extension/archetype/it/ExtensionArchetypeGenerationTestCase.java

```diff
@@ -89,10 +95,28 @@ private void generate(String groupId, String artifactId, String version, Propert
       cliProps.add("-s " + System.getProperty(MAVEN_SETTINGS_PROPERTY, (String) null));
       verifier.setCliOptions(cliProps);
     }
-
+    verifier.addCliOption("-DskipDocumentation");
     verifier.setMavenDebug(true);
-    verifier.executeGoals(asList("compile", "test"), getEnvVars());
+    verifier.executeGoals(asList("package"), getEnvVars());
     verifier.verifyErrorFreeLog();
+    Optional<String> extensionName = Optional.ofNullable(pluginProperties.getProperty(EXTENSION_NAME));
+    verifyExtensionModel(extensionName.orElse(TEST_EXTENSION_NAME), artifactId);
+  }
+
+  private void verifyExtensionModel(String extensionName, String artifactId) throws Exception {
+    String normalizedExtensionName = extensionName.toLowerCase().replace(" ", "-");
+    String actualExtensionModel = getActualExtensionModel(artifactId);
+    String expectedExtensionModel = getExpectedExtensionModel(normalizedExtensionName, artifactId) ;
+    JSONAssert.assertEquals(expectedExtensionModel, actualExtensionModel,true);
+  }
+
+  private String getExpectedExtensionModel(String extensionName, String artifactId) throws Exception {
+    return IOUtils.getResourceAsString("./src/test/resources/"+extensionName+".json",this.getClass());
```

Comment:
InsputStream is = Thread.currentThread().getContextClassLoader().gerResourceAsStream(extensionName+".json");
return IOUtils.toString(is);